src/kani-compiler/src/codegen_cprover_gotoc/codegen/place.rs

```diff
@@ -550,3 +584,39 @@ impl<'tcx> GotocCtx<'tcx> {
         arr.member("0", &self.symbol_table).index_array(idx)
     }
 }
+
+/// A convenience macro that unwraps a `Result<ProjectPlace<'tcx>,
+/// Err<UnimplementedData>` if it is Ok and returns an `codegen_unimplemented`
+/// expression otherwise
```

Comment:
quick question: why macro and not function?

Reply:
For this to have minimal changes on the caller side, it has to do a `return`, which cannot be done using a function (since it will return from that function, and not from the caller function). I hope the explanation is not too confusing.